lms/djangoapps/django_comment_client/management/tests.py

```diff
@@ -0,0 +1,245 @@
+        self.assertRegexpMatches(result['message'], 'does not exist')
+
+    def test_rename_failed(self):
+        # ensures that we throw an error if two users have the same username
+        args = [self.sql_users[0].username, self.sql_users[1].username]
+        result = management_utils.rename_user(*args)
+        self.assertRegexpMatches(result['message'], 'Error modifying the username for')
+
+    def test_change_username(self):
+        # ensure that the starting sql username is different from the target username
+        self.assertTrue(self.sql_users[0].username != TestRenameUser.NEW_USERNAME)
+
+        # ensure that the starting mongo username matches the starting sql username
+        mongo_user = self.db.users.find_one({'username': self.sql_users[0].username})
+        self.assertTrue(mongo_user['username'] == self.sql_users[0].username)
+        mongo_comment = self.db.contents.find_one({'author_username': self.sql_users[0].username})
+        self.assertTrue(mongo_comment['author_username'] == self.sql_users[0].username)
+
+        # perform the rename
+        args = [self.sql_users[0].username, TestRenameUser.NEW_USERNAME]
+        result = management_utils.rename_user(*args)
+
+        self.assertEqual(result['status'], 'success')
+        # check that the sql username now matches the target username
+        self.sql_users[0] = User.objects.get(id=self.sql_users[0].id)
+        self.assertEqual(self.sql_users[0].username, TestRenameUser.NEW_USERNAME)
+
+        # check that the mongo username now matches the target username
+        mongo_user = self.db.users.find_one({'_id': mongo_user['_id']})
+        self.assertEqual(mongo_user['username'], TestRenameUser.NEW_USERNAME)
+
+        mongo_comment = self.db.contents.find_one({'_id': mongo_comment['_id']})
+        self.assertEqual(mongo_comment['author_username'], TestRenameUser.NEW_USERNAME)
+
+    def test_change_username_not_matching(self):
+        # ensure that the username for the 2nd user does not change when renaming the first user
+        self.assertNotEqual(self.sql_users[1].username, TestRenameUser.NEW_USERNAME)
+        mongo_user2 = self.db.users.find_one({'username': self.sql_users[1].username})
+        mongo_comment2 = self.db.contents.find_one({'author_username': self.sql_users[1].username})
+        self.assertEqual(mongo_user2['username'], self.sql_users[1].username)
+        self.assertEqual(mongo_comment2['author_username'], self.sql_users[1].username)
+
+        # perform the rename only for the first user
+        args = [self.sql_users[0].username, TestRenameUser.NEW_USERNAME]
+        result = management_utils.rename_user(*args)
+        # ensure that another user was successfully changed
+        self.assertEqual(result['status'], 'success')
+
+        # re-query the second user
+        self.sql_users[1] = User.objects.get(id=self.sql_users[1].id)
+        mongo_user2 = self.db.users.find_one({'_id': mongo_user2['_id']})
```

Comment:
maybe some sort of assertExists or assertTrue just to make sure the `find_one` succeeded?